### ISSUE TITLE
React 17

### ISSUE THREAD

**Issue by birge** [airbnb/react-with-direction#20]
React 17 is around the corner and this package still relies on the unsafe `componentWillRecieveProps`. I attempted to make a pr to mark it as unsafe but this breaks with pre 16 react versions. Can we move forward and remove support for pre 16 react? If not switching to `componentDidUpdate` in this PR https://github.com/airbnb/react-with-direction/pull/14 seems like the logical way forward.

**Comment by ljharb** [comment]
I'm not sure how accurate "around the corner" is.

As for #14, see https://github.com/airbnb/react-with-direction/pull/14#discussion_r195531197.

**Comment by mrbalov** [comment]
Do we expect to have React 17 support?

**Comment by ljharb** [comment]
I do expect to have it.

**Comment by mcmxcdev** [comment]
Landing here due to incorrect peer dependenciy warnings from `yarn install` for `react` and `react-dom`, would be great to get this resolved!

**Comment by ljharb** [comment]
@mhatvan they're currently correct warnings.

**Comment by nbstr** [comment]
Is this on the roadmap guys?

`npm ERR! Found: react@17.0.2
npm ERR! node_modules/react
npm ERR!   react@"^17.0.2" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^0.14 || ^15 || ^16" from react-with-direction@1.3.1`

react@17 unsupported..

**Comment by JustFly1984** [comment]
Please add support for react 17

**Comment by mrbalov** [comment]
> @mhatvan they're currently correct warnings.

Yeah, the warning is correct 😄 And it should be solved by the `airbnb` team by updating `peerDependency` at `package.json`.

It's a sad thing that so huge a company doesn't care about it.

**Comment by ljharb** [comment]
@balovbohdan thats hostile, untrue, and unnecessary. Everybody cares a lot, which is why we won’t update the peer dep until we’re able to test it. It would be reckless to do so otherwise.

**Comment by mrbalov** [comment]
https://github.com/airbnb/react-with-direction/pull/32

**Comment by mrbalov** [comment]
> @balovbohdan thats hostile, untrue, and unnecessary. Everybody cares a lot, which is why we won’t update the peer dep until we’re able to test it. It would be reckless to do so otherwise.

Yep, you're right. But this may be to so complex as expected. React 18 is rather close to getting launched, and folks are still not able to migrate to React 17. It's not correct to go with this warning, as it means dependency is not correct and something may get broken.

**Comment by ljharb** [comment]
Correct. Your choices remain to stay on react 16, or stop using this library.
